File: pa5/DictionaryTest.c
//-----------------------------------------------------------------------------
// DictionaryTest.c
// Test client for the Dictionary ADT
//-----------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

int main(int argc, char* argv[]){
	Dictionary A = newDictionary();
	char* k;
	char* v;
	char* array1[] = {"one","two","three","four","five","six","seven","eight","nine","ten"};
	char* array2[] = {"difference","always","memory","galumph","happy","sad","results","assignment","storing", "strncat"};
	printf("Insert one key and value:\n");
    insert(A,array1[0],array2[0]);
    printf("A is empty? %s\n", (isEmpty(A)?"true":"false"));
    printf("Size of A is: %d\n", size(A));
    printDictionary(stdout, A);
    makeEmpty(A);
    printf("Make empty A\n");
    printf("A is empty? %s\n", (isEmpty(A)?"true":"false"));
    printf("Size of A is: %d\n", size(A));

    printf("Insert 10 keys and values\n");
    for(int i=0; i<10; i++){
      insert(A, array1[i], array2[i]);
    }
    printf("A is empty? %s\n", (isEmpty(A)?"true":"false"));
    printf("Size of A is: %d\n", size(A));

    for(int i=0; i<10; i++){
      k = array1[i];
      v = lookup(A, k);
      printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
    }

    delete(A, "one");
    delete(A, "five");
    delete(A, "ten");
    // delete(A, "one");  // error: key not found
    printf("Delete the keys one, five and ten...\n");
    printf("A is empty? %s\n", (isEmpty(A)?"true":"false"));
    printf("Size of A is: %d\n", size(A));
    for(int i=0; i<10; i++){
      k = array1[i];
      v = lookup(A, k);
      printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
    }
    
    makeEmpty(A);
    printf("Make empty A\n");
    printf("A is empty? %s\n", (isEmpty(A)?"true":"false"));
    printf("Size of A is: %d\n", size(A));

    freeDictionary(&A);
    return(EXIT_SUCCESS);



}

File: pa3/DictionaryTest.java
//-----------------------------------------------------------------------------
// DictionaryTest.java
// Client module for testing Dictionary ADT
//-----------------------------------------------------------------------------

public class DictionaryTest{

   public static void main(String[] args){
      String v;
      Dictionary A = new Dictionary();
      A.insert("1","a");
      System.out.println(A);
      A.delete("1");
      System.out.println(A);
      A.insert("1","a");
      A.insert("2","b");
      System.out.println(A);
      A.delete("1");
      System.out.println(A);
      A.insert("1","a");
      System.out.println(A);
      A.delete("2");
      System.out.println(A);
      System.out.println("isEmpty ?" + A.isEmpty());
      System.out.println("size : " + A.size());
      A.makeEmpty();
      System.out.println();
      A.insert("1","a");
      A.insert("2","b");
      A.insert("3","c");
      A.insert("4","d");
      A.insert("5","e");
      A.insert("6","f");
      A.insert("7","g");
      A.insert("8","peace");
      System.out.println(A);

      try{
         A.insert("2","f");  // causes DuplicateKeyException
      }catch(DuplicateKeyException e){
         System.out.println("Caught Exception " + e);
         System.out.println("Continuing without interuption");
      }

      v = A.lookup("1");
      System.out.println("key=1 "+(v==null?"not found":("value="+v)));
      v = A.lookup("4");
      System.out.println("key=3 "+(v==null?"not found":("value="+v)));
      v = A.lookup("6");
      System.out.println("key=7 "+(v==null?"not found":("value="+v)));
      v = A.lookup("8");
      System.out.println("key=8 "+(v==null?"not found":("value="+v)));
      v = A.lookup("9");
      System.out.println("key=9 "+(v==null?"not found":("value="+v)));
      System.out.println();

      
      A.delete("1");
      A.delete("4");
      A.delete("6");
      System.out.println(A);

      try{
         A.delete("6");  // causes KeyNotFoundException
      }catch(KeyNotFoundException e){
         System.out.println("Caught Exception " + e);
         System.out.println("Continuing without interuption");
      }

      System.out.println("isEmpty ?" + A.isEmpty());
      System.out.println("size : " + A.size());
      A.makeEmpty();
      System.out.println("isEmpty ?" + A.isEmpty());
      System.out.println(A);

   }
}